public/app/features/dashboard-scene/sharing/ShareSnapshotTab.tsx

```diff
@@ -1,20 +1,212 @@
+
+    const cmdData = {
+      dashboard: snapshot,
+      name: snapshot.title,
+      expires: selectedExpireOption?.value,
+      external: false,
+    };
+
+    try {
+      const results: { deleteUrl: string; url: string } = await getBackendSrv().post(SNAPSHOTS_API_ENDPOINT, cmdData);
+      console.log(results);
+      console.log(results.url);
+      // this.setState({
+      //   deleteUrl: results.deleteUrl,
+      //   snapshotUrl: results.url,
+      //   step: 2,
+      // });
+    } finally {
+      // trackDashboardSharingActionPerType(external ? 'publish_snapshot' : 'local_snapshot', shareDashboardType.snapshot);
+      // this.setState({ isLoading: false });
+    }
+  };
+
+  public onExternalSnapshotCreate = () => {};
+}
+
+function ShareSnapshoTabRenderer({ model }: SceneComponentProps<ShareSnapshotTab>) {
+  const { snapshotName, selectedExpireOption, timeout, modalRef, snapshotSharingOptions } = model.useState();
+
+  // const [s, fn] = useAsyncFn(async () => {
+  //   model.onLocalSnapshotCreate();
+  // });
+  return (
+    <>
+      <div>
+        <p className="share-modal-info-text">
+          <Trans i18nKey="share-modal.snapshot.info-text-1">
+            A snapshot is an instant way to share an interactive dashboard publicly. When created, we strip sensitive
+            data like queries (metric, template, and annotation) and panel links, leaving only the visible metric data
+            and series names embedded in your dashboard.
+          </Trans>
+        </p>
+        <p className="share-modal-info-text">
+          <Trans i18nKey="share-modal.snapshot.info-text-2">
+            Keep in mind, your snapshot <em>can be viewed by anyone</em> that has the link and can access the URL. Share
+            wisely.
+          </Trans>
+        </p>
+      </div>
+
+      <Field label={t('share-modal.snapshot.name', `Snapshot name`)}>
+        <Input
+          id="snapshot-name-input"
+          width={30}
+          defaultValue={snapshotName}
+          onBlur={(e) => model.onSnasphotNameChange(e.target.value)}
+        />
+      </Field>
+
+      <Field label={t('share-modal.snapshot.expire', `Expire`)}>
+        <RadioButtonGroup<number>
+          id="expire-select-input"
+          options={EXPIRE_OPTIONS}
+          value={selectedExpireOption?.value}
+          onChange={model.onExpireChange}
+        />
+      </Field>
+
+      <Field
+        label={t('share-modal.snapshot.timeout', `Timeout (seconds)`)}
+        description={t(
+          'share-modal.snapshot.timeout-description',
+          `You might need to configure the timeout value if it takes a long time to collect your dashboard metrics.`
+        )}
+      >
+        <Input
+          id="timeout-input"
+          type="number"
+          width={21}
+          value={timeout}
+          onChange={(e: React.ChangeEvent<HTMLInputElement>) => model.onTimeoutChange(Number(e.target.value))}
+        />
+      </Field>
```

Comment:
we don't need this option :)

Reply:
I know, started the work with the quick UI, hence I left it here :D